hwilib/commands.py

```diff
@@ -157,21 +165,40 @@ def getdescriptor(client, master_xpub, testnet=False, path=None, internal=False,
     if client.xpub_cache.get(path_base) is None:
         client.xpub_cache[path_base] = client.get_pubkey_at_path(path_base)['xpub']
 
-    return Descriptor(master_fpr, path_base.replace('m', ''), client.xpub_cache.get(path_base), path_suffix, client.is_testnet, sh_wpkh, wpkh)
+    return Descriptor(master_fpr, path_base.replace('m', ''), client.xpub_cache.get(path_base), path_suffix, client.is_testnet, is_sh_wpkh, is_wpkh)
+
+def getkeypool(client, path, start, end, internal=False, keypool=True, account=0, sh_wpkh=False, wpkh=True, addr_all=False):
 
-# wrapper to allow both internal and external entries when path not given
-def getkeypool(client, path, start, end, internal=False, keypool=True, account=0, sh_wpkh=False, wpkh=True):
+    if sh_wpkh and wpkh:
+        return {'error': 'Both `--wpkh` and `--sh_wpkh` can not be selected at the same time.', 'code': BAD_ARGUMENT}
+    if addr_all and (sh_wpkh or wpkh or (path is not None) or internal):
+        return {'error': 'Both `--all` and specific address/change types can not be selected at the same time.', 'code': BAD_ARGUMENT}
```

Comment:
Instead of having the checks like this and requiring multiple bool args, we should just add a mutually exclusive group to the argument parser itself. That should do what we want and should be able to get it to give us the correct enum too.

Reply:
Still punting enum list construction until `getkeypool` internally, but otherwise done(I think?)